cpp/src/community/detail/common_methods.cuh

```diff
@@ -183,10 +187,67 @@ weight_t compute_modularity(
   return Q;
 }
 
+template <typename map_key_t, typename map_value_t, bool multi_gpu>
+rmm::device_uvector<map_value_t>&& lookup_primitive_values_for_keys(
+  raft::handle_t const& handle,
+  rmm::device_uvector<map_key_t>& map_keys,
+  rmm::device_uvector<map_value_t>& map_values,
+  rmm::device_uvector<map_key_t>& keys_to_lookup)
+{
+  static_assert(std::is_integral<map_key_t>::value);
+  static_assert(std::is_same<map_key_t, std::int32_t>::value ||
+                std::is_same<map_key_t, std::int64_t>::value);
```

Comment:
I assume `static_assert(std::is_integral<map_key_t>::value);` is redundant here.